### DISCUSSION TITLE
My first evy - this is great fun!

### DISCUSSION THREAD

**Discussion by keilin-anz** [evylang/evy#224] [Show and tell]
https://gist.github.com/keilin-anz/54a56bcb58c642aa7ebf9438674f324a#file-chaser-evy

[at evy.dev/play](https://evy.dev/#content=H4sIAAAAAAAAA31VS2/bMAy++1cQAQKkDYK4HbpDsexSbLedVmzxUbXpRqgtBZKcJvv1I/Ww5RSoEdgi+fH1iVK222K7hV96sAj1QVg0JPOP1ZvNBn6jG45Qa2WdUM6yzj+FPSI28LiD+7Jojayd1IrFh8nVCYejw/QUIdGe0WUUqlz4g93+So72wgnzii5ag5BMlBVPFzgKa9GCUCBerO4GKsHJHsEId0AD7kAWr2iwc4K9rIZ3BOu0IegB4WjwJImRAJMKtGnYU0MtunrouKvgnJDPDByr4N7xhDlbU+8FsdTrE8L5UQ09XPhdAD2psx2cM7ki+VKgaryjULLn7G70onRXRXEt3jRKXNnKwQbycm9gC3dlWXrorI8duCLFpk6GY8OxV56MFz2o5sZbo37Kmfk0RrzDqjVi7lLrjmuxR6medKeJ0qDuUBg2hrLJ9682XRO6jnwyZT1RGtMmZouiHVQ91vI88mKJFWx+ZnN5Rw2v7mBNKLiFNLGxsGnsdrlwexVnDq5ycPURPKGzpZlS7KmYWbLmOcNN0ascVwWc58b3HhJ45qThRumDoeuUJoVJIxZHK1TFzPrzVGvLvqPSnytL48/KT2laxyDEAF8Kn7K0jrETNt9k3nqpXq9PTWx0Gow0Jj+Uk+4CC4PNAhaNMG9+dd32Q8ZW5kYD92id4YT00W/4NCmyw0kU/MNxrlrZdeOoBrfJe6QOznRoPQvS0HT7ELNOByc76STOb4hY43g+UjUNvo4F0JoI5fcSvnwt5+eK0K6FxcF2q2UL5XLJR5zeW7hfLm8W7OYdDF3qRvk+skHiO94IqXzX8dNLunKGPqzFOa19FNny5fgtYbwui56rsaP/l4D/nuJ8wGdqriozSZVvYZpvX1S+U07v47fK+KL7yc83WekOPE/aKmgr0l7ybIL+6e4TJPj79P8BxmR2riwHAAA=)

**Comment by juliaogris** [comment]
@keilin-anz - this is super cool!
Would you mind if I add it to the gallery, and maybe also to the examples?
(I will of course attribute you).

**Reply by pentaphobe** [reply]
answered elsewhere, but of course - please assume the same for anything else I write in evy :D

**Comment by pentaphobe** [comment]
Just discovered the slider inputs and had to try this:

## L-System example

![7ce28263-e50f-4844-b926-1a2193f907f4](https://github.com/evylang/evy/assets/418523/8a9142ae-3a3e-4b4c-a2f3-7ebed810e948)


[gist](https://gist.github.com/pentaphobe/31dddbd8554bab1f266924d764fa1a79)
 [on evy.dev/play](https://evy.dev/#content=H4sIAAAAAAAAA4VXTW7jNhTe+xRvXAxgJ4gjJ/FMYjQFBrPqoijQAfqDwAvaomzCMiWQVBx3MGfovqseo+fpBXqFvh+KkjIzjRYSaX7v4+P7pWsDy3u4ns1v5ou70ejyEi4uLsAH5YKxW6grb4KpLCibgze/a6gKCDsNwWlN6OAau4dS223YjVjsA6GIM6P1ujKWmZr6qFzuXwnond2WjHIqN8p6uMu6zWu9MYXRPm0Em6qsGudHrqrCexw7Ev24a/TyagFeheWbGyjNdheWV9mnUTD1EHQnoNu7CLp+8yntFnZms7fa+/Zka6fsZqdls19MHnZ8GmJNs2x2tUgMm8oGV5Wirt+p+jMj7aojHJrNDrTCl2wAoXF47sJVBzDBQ62ctmGkyDDvd/gZmOdmMeDxB1WW2g34DCmgbJ+MHNZxZbM3119R+qhO5KPnvsXxpnFEBawXAQgrWv5YFF4HZm6VK5QPcBTCEb0+1FrnsvltB3IQqg7z7lA1NkSzDo5ZKrftHCIS/NsHHIlA9gXDPJPg31qJq2SB0hzIVCyq7KkTQt1yp46E++fPv/79+w9Yox3QokVTwobMSbbCuMFgPqin74PmSFsACuDcHJoDjHNdh92YDJbiiSwaqQ/GtmmSzUTQWBaUTCI5FSVHo1o1XqxYqNLr0Yjz0RxUQF8tbXMYAT6mgFeC5Ck9fTfdw8QbC5MAlzDPsgzOIDloOo0zcQXLa5uPeOA0q8zzorGb9gda25QanTn+Rl8Xt8XbMf925CTBYwmCE3GMMrJaGPTP2FZWj4X+UD1qWGSoE0+JWqbQKxRdYclEEbQYIX2yEhcoDFgbfyfXhmOF/jJljhEsqjP7E5kMTvxmC/GIcoUHBv3ZNypXvW+hdRmehhDwHUTfJ2s7TSk9NB6Xx1Q2nD6oWggfWr50tBU8pBKTSs+qZ9KOS7h9DOmXWLv4T7mzEgaVMk84+GAPWXuyFY3nq54fJ/z5wTxBV4lTtRW2qVBbffyViJ/gnKS8WJrjDLVqIb8R5EQQis0hhDHRz5isVkNBu2HatYlBAF7gzYgu7Z36i9Qt3KCfCxLs0xbcpiKbUHaXqjnUgcphqYsgMdVXgpeTEmn/i34xb/chG5/PPyN21Jhms9n/8Z2/zMfZtFabPZUaLu0UCF2iPcEpJRC2gEftQuxBud5i2efqF5tOzHaZcGpETMoOiXiYtLJnUJsp1ZfbLk2x/WtXVyUVK96oIoc1h7XmLqBshWo6QsZoND51ibUOR40prTi719RZtjogluFUxb/+wJpAGfQeKrXPnrkcUpSUisDvtYxZo3TcXJdBUaCs4QJU3wKKIkyWz1IedHaOOYMnK81ex91g3QQO6fZyQ8YKsMYOkPt0dO7HWMdqVz2aHLtARKOTqSdQJ5CaTRZBI4lCVBd9cNSnTlWDzcsCNgasI1hGWJ3JVA7e6raMaLX8+EnOL99nFuAqFtVXM7xawZrfqusbfgDBaxdC6N2DlAMIX8kQJN8eLNrW16hZKGC88+Xk9azAYAP8vH4t7+kYtULDPg9rtECMshR1jhNnGHR6O4vxBLEILrjqZVj8uA1lq7h+rBrMelZpQdpX8HbRNkSUk95h4+cn2mr5sKJZ1YTeNB7NN2UQM2AsRfxDtuqv3reDS5i0iPlqAJ9+EX8Gk3ZPEUizr0mc9yF942Mn6C/BBBsM9MmFIAY7RuNen2Afo6ntoHu4v4cxjFOnjBea+8GFhRhaFmNrTA6Tt2H5qMpnnCZnUl9iUrinjjqWop9V2eiusU1Q+Ip8gURT8S87GP92rIa3pVhdu5t3n1D0pCtYK4Nx0btkv7BdNrtZ3GIVuovtl9MlNecXhRewWA0t9R/+1u1svQ0AAA==)

**Reply by pentaphobe** [reply]
Bonus content for anyone who discovers this later, try using [Cubic Interpolation](https://easings.net/#easeInOutCubic) instead of [Linear Interpolation](https://en.wikipedia.org/wiki/Linear_interpolation) with this drop in snippet:

```gml
// interpolate from one number to another
// amount is how far between a and b we get
//
// a ----------------------------- b
// 0            0.5                1
func interp:num a:num b:num amount:num
    delta := b - a

    // cubic easing makes everything prettier!
    // https://easings.net/#easeInOutCubic
    if amount < 0.5
        amount = 4 * amount * amount * amount
    else
        amount = (1 - (pow (-2 * amount + 2) 3)) / 2
    end

    return a + (delta * amount)
end
```
